backend/src/server/plugins/error-handler.ts

```diff
@@ -36,6 +44,36 @@ export const fastifyErrHandler = fastifyPlugin(async (server: FastifyZodProvider
         status: error.status,
         detail: error.detail
       });
+      // Handle JWT errors and make them more human-readable for the end-user.
+    } else if (error instanceof JsonWebTokenError) {
+      // We wan't to return a slightly different message if the request is coming from the CLI. This is because we're able to provide more specific information to the user in that case.
+      const isCliRequest = req.headers["user-agent"] === UserAgentType.CLI;
+
+      const message = (() => {
+        if (error.message === JWTErrors.JwtExpired) {
+          return "Your token has expired. Please re-authenticate.";
+        }
+        if (error.message === JWTErrors.JwtMalformed) {
+          if (isCliRequest) {
+            return "The access token is malformed. Are you sure the token you are using is correct? Check the INFISICAL_TOKEN environment variable, or the --token flag. If you are using user-login, please run [infisical login] to re-authenticate. Please ensure that the INFISICAL_TOKEN variable is not set, if you are not intentionally using it.";
+          }
+          return "The access token is malformed. Please ensure that the token is in the correct format and try again.";
+        }
+        if (error.message === JWTErrors.InvalidAlgorithm) {
+          if (isCliRequest) {
+            return "Invalid algorithm. Are you sure you are using the correct authentication method? Make sure to check that you don't have the INFISICAL_TOKEN variable set in your environment variables. If you are intentionally using the INFISICAL_TOKEN variable, make sure you are using the correct token.";
```

Comment:
Instead of this, we can add a warning in the CLI when the access token is being passed from `INFISICAL_TOKEN`. This way the user knows that their current session is being over written